### PR TITLE
K8SPSMDB-387 feature: add restore to the latest chunk

### PR DESCRIPTION
**Pull request by ollevche**
[K8SPSMDB-387](https://jira.percona.com/browse/K8SPSMDB-387) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-387

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| security-context | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/7f0ed15f3f68fb73a9d92c9ce116f84d91fa3353
image: `perconalab/percona-server-mongodb-operator:PR-626-7f0ed15f`

File: pkg/psmdb/backup/pbm.go [hunk 1]
```diff
@@ -200,3 +200,12 @@ func (b *PBM) HasLocks(predicates ...LockHeaderPredicate) (bool, error) {
...
 	return false, nil
 }
+
+func (b *PBM) GetLastPITRChunk() (*pbm.PITRChunk, error) {
+	masterNode, err := b.C.GetNodeInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting master node")
+	}
+
+	return b.C.PITRLastChunkMeta(masterNode.SetName)
```
Comment (delgod) [inline pkg/psmdb/backup/pbm.go:210]: 1) it is wrong to get last chunk from database itself. we should restore to the latest chunk on the storage.
again: think about case `clusterA makes backups to storageA, clusterB makes backups to storageB.
now we need to restore PITR backup from storageA to clusterB to the latest backed up chunk`

2) `master` terminology can be sensitive, please keep using netral terminology here. e.g.: `nodeInfo`

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/psmdb/backup/pbm.go [hunk 2]
```diff
@@ -200,3 +200,12 @@ func (b *PBM) HasLocks(predicates ...LockHeaderPredicate) (bool, error) {
 
 	return false, nil
 }
+
+func (b *PBM) GetLastPITRChunk() (*pbm.PITRChunk, error) {
+	nodeInfo, err := b.C.GetNodeInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting master node")
```
Comment (delgod) [inline pkg/psmdb/backup/pbm.go:207]:
```suggestion
		return nil, errors.Wrap(err, "getting node information")
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| storage | passed |
| init-deploy | passed |
| service-per-pod | passed |
| limits | passed |
| liveness | passed |
| self-healing | passed |
| scaling | passed |
| smart-update | passed |
| operator-self-healing | passed |
| security-context | passed |
| version-service | passed |
| rs-shard-migration | passed |
| users | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/7f0ed15f3f68fb73a9d92c9ce116f84d91fa3353
image: `perconalab/percona-server-mongodb-operator:PR-626-7f0ed15f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/dcd8bc94b815a9fa1efd4dc7d700e3276dd24b69
image: `perconalab/percona-server-mongodb-operator:PR-626-dcd8bc94`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/2b0f3851160b4914fd1e75fee0f9a18f55a142cd
image: `perconalab/percona-server-mongodb-operator:PR-626-2b0f3851`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/2b0f3851160b4914fd1e75fee0f9a18f55a142cd
image: `perconalab/percona-server-mongodb-operator:PR-626-2b0f3851`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| one-pod | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/2b0f3851160b4914fd1e75fee0f9a18f55a142cd
image: `perconalab/percona-server-mongodb-operator:PR-626-2b0f3851`

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| security-context | passed |
| self-healing | passed |
| rs-shard-migration | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/2b0f3851160b4914fd1e75fee0f9a18f55a142cd
image: `perconalab/percona-server-mongodb-operator:PR-626-2b0f3851`

**Review by blez — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/626/commits/23e2fc620d153ab9f000f900f376e0fc5971deee
image: `perconalab/percona-server-mongodb-operator:PR-626-23e2fc62`

**Pull request merged**